geomesa-core/src/main/scala/org/locationtech/geomesa/core/iterators/QuerySizeIterator.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2015 Commonwealth Computer Research, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.locationtech.geomesa.core.iterators
+
+import java.util
+
+import org.apache.accumulo.core.data._
+import org.apache.accumulo.core.iterators.{IteratorEnvironment, SortedKeyValueIterator}
+import org.geotools.feature.simple.SimpleFeatureBuilder
+import org.locationtech.geomesa.core
+import org.locationtech.geomesa.core.iterators.QuerySizeIterator._
+import org.locationtech.geomesa.feature.{FeatureEncoding, SimpleFeatureEncoder}
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
+
+/**
+ * Iterator that returns counts and size-in-bytes of both results filtered and results returned.
+ */
+class QuerySizeIterator extends GeomesaFilteringIterator with HasFeatureDecoder with HasFilter with HasFeatureType {
+
+  var featureBuilder: SimpleFeatureBuilder = null
+  var querySizeFeatureEncoder: SimpleFeatureEncoder = null
+
+  override def init(source: SortedKeyValueIterator[Key, Value], options: util.Map[String, String], env: IteratorEnvironment): Unit = {
+    super.init(source, options, env)
+    initFeatureType(options)
+    // Need to grab the original SFT in order to properly decode/filter
+    val originalSFT = SimpleFeatureTypes.createType("QuerySizeHijackedIteratorSFT",options.get(core.GEOMESA_ITERATORS_SIMPLE_FEATURE_TYPE))
+    super[HasFilter].init(originalSFT, options)
+    super[HasFeatureDecoder].init(originalSFT, options)
```

Comment:
I can get HasFilter implicitly (iirc) but I had to specify super for HasFeatureDecorder, I figured specifying both would improve readability, should I remove the super reference that it finds through linearization?